cpp/src/arrow/util/value_parsing.cc

```diff
@@ -53,8 +53,11 @@ bool StringToFloat(const char* s, size_t length, char decimal_point, uint16_t* o
   float temp_out;
   const auto res =
       ::arrow_vendored::fast_float::from_chars_advanced(s, s + length, temp_out, options);
-  *out = Float16::FromFloat(temp_out).bits();
-  return res.ec == std::errc() && res.ptr == s + length;
+  const bool ok = res.ec == std::errc() && res.ptr == s + length;
```

Comment:
`res.ptr == s + length` changes the logic for parsing?

Reply:
No, it's just that otherwise the result is a failure and there is no need to do the final conversion to float16.